services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java

```diff
@@ -121,6 +125,10 @@ public Response performLogin(AuthenticationRequest request) {
                 authnRequest = it.next().beforeSendingLoginRequest(authnRequest, request.getAuthenticationSession());
             }
 
+            destinationUrl = ofNullable(authnRequest.getDestination())
+                    .map(URI::toString)
+                    .orElseThrow(() -> new IdentityBrokerException("No destination in login request"));
+
```

Comment:
There should be no exception thrown. The destination is attribute only mandatory in certain cases, see the SAML spec above.
The code should be similar to:
```java
if (authnRequest.getDestination() != null) {
    destinationUrl = authnRequest.getDestination().toString();
}
```

Similar code needs to be in `keycloakInitiatedBrowserLogout` and `backchannelLogout` methods.